src/index.ts

```diff
@@ -1,273 +1,95 @@
+import {ParseOptions, pathToRegexp, TokensToRegexpOptions} from './path-to-regex-6.2';
 import {
-  pathToRegexp,
-  TokensToRegexpOptions,
-  ParseOptions
-} from "./path-to-regex-6.2";
+  canonicalizeHash,
+  canonicalizeHostname,
+  canonicalizePassword,
+  canonicalizePathname,
+  canonicalizePort,
+  canonicalizeProtocol,
+  canonicalizeSearch,
+  canonicalizeUsername,
+  defaultPortForProtocol,
+  extractValues,
+  isAbsolutePathname,
+} from './url-pattern-utils';
```

Comment:
I think I would just call this url-utils as it is really that generic without extractValues